peer/ping_manager.go

```diff
@@ -143,53 +177,65 @@ func (m *PingManager) pingHandler() {
 
 			// Set up our bookkeeping for the new Ping.
 			if err := m.setPingState(pongSize); err != nil {
-				m.cfg.OnPongFailure(err)
+				// This is an internal error related to timer
+				// reset. Pass it to OnPongFailure as it's
+				// critical. Current and last RTT are not
+				// directly applicable here.
+				m.cfg.OnPongFailure(err, 0, 0)
 
-				return
+				continue
```

Comment:
I turned this into a `continue` as well, since we don't necessarily disconnect. Otherwise if we ran into this but the user has configured to not disconnect, then the ping manager won't be running anymore but the peer is still connected (which could have weird side effects).

Reply:
do we need to reset the state here too?